### PR TITLE
Yautja Vision Changes

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request
This change alters Yautja to fit with the lore on our wiki, giving them natural thermal vision, and a weak night vision capability. To counterbalance this, in order to see clearer with the normal NVGs (which also allow to see the turfs behind other walls), a 5 charge per process fee must be paid, which when operating on the colony results in no natural recharge unless visor is disabled.
I have removed the termal and meson views from the clan mask, as meson was obsolete already and thermal now integrated with species. In doing said integration I have also added a var so different species can have differing inherent sight flags.
I've also updated the checks on the mask for using the visor, as previously you could not change visor setting without bracer, but you could keep the setting of a stolen mask without a bracer.

# Explain why it's good for the game
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
fix: Picking up a Yautja mask now correctly checks for bracers to apply the visor.
add: Added backend for species datums to hold inherent vision flags.
add: Added proc for Yautja mask to pass power drain back to bracers. Added small power drain on use of mask visor.
add: Added natural thermal vision and very weak night vision to Yautja species, to fit with our wiki lore.
del: Removed Thermal and Meson visors from Yautja mask.
/:cl:

### CODE REVIEW

File: code/modules/cm_preds/yaut_mask.dm [hunk 1]
```diff
@@ -65,8 +63,33 @@
 
 /obj/item/clothing/mask/gas/yautja/Destroy()
 	remove_from_missing_pred_gear(src)
+	STOP_PROCESSING(SSobj, src)
 	return ..()
 
+/obj/item/clothing/mask/gas/yautja/process()
+	if(!ishuman(loc))
+		STOP_PROCESSING(SSobj, src)
```
Comment (harryob) [inline code/modules/cm_preds/yaut_mask.dm:71]:
```suggestion
```

File: code/modules/cm_preds/yaut_mask.dm [hunk 2]
```diff
@@ -65,8 +63,33 @@
...
+		return
```
Comment (harryob) [inline code/modules/cm_preds/yaut_mask.dm:72]:
```suggestion
		return PROCESS_KILL
```

File: code/modules/cm_preds/yaut_mask.dm [hunk 3]
```diff
@@ -65,8 +63,33 @@
...
+	var/mob/living/carbon/human/human_holder = loc
+
+	if(current_goggles && !drain_power(human_holder, 5))
+		to_chat(human_holder, SPAN_WARNING("Your bracers lack sufficient power to operate the visior."))
+		current_goggles = VISION_MODE_OFF
+		var/obj/item/visor = human_holder.glasses
+		if(istype(visor,/obj/item/clothing/glasses/night/yautja))//To change if any new vision modes are made
```
Comment (harryob) [inline code/modules/cm_preds/yaut_mask.dm:79]:
```suggestion
		if(istype(visor, /obj/item/clothing/glasses/night/yautja))//To change if any new vision modes are made
```

File: code/modules/cm_preds/yaut_mask.dm [hunk 4]
```diff
@@ -91,13 +114,13 @@
...
 		return
-	var/obj/item/clothing/gloves/yautja/Y = M.gloves //Doesn't actually reduce power, but needs the bracers anyway.
+	var/obj/item/clothing/gloves/yautja/Y = M.gloves
 	if(!Y || !istype(Y))
 		to_chat(M, SPAN_WARNING("You must be wearing your bracers, as they have the power source."))
 		return
 	var/obj/item/G = M.glasses
 	if(G)
-		if(!istype(G,/obj/item/clothing/glasses/night/yautja) && !istype(G,/obj/item/clothing/glasses/meson/yautja) && !istype(G,/obj/item/clothing/glasses/thermal/yautja))
+		if(!istype(G,/obj/item/clothing/glasses/night/yautja))
```
Comment (harryob) [inline code/modules/cm_preds/yaut_mask.dm:123]:
```suggestion
		if(!istype(G, /obj/item/clothing/glasses/night/yautja))
```

File: code/modules/cm_preds/yaut_mask.dm [hunk 5]
```diff
@@ -112,49 +135,47 @@
...
 			var/datum/mob_hud/H = huds[listed_hud]
 			H.remove_hud_from(user)
-		var/obj/item/G = user.glasses
-		if(G) //make your hud fuck off
-			if(istype(G,/obj/item/clothing/glasses/night/yautja) || istype(G,/obj/item/clothing/glasses/meson/yautja) || istype(G,/obj/item/clothing/glasses/thermal/yautja))
-				user.temp_drop_inv_item(G)
-				qdel(G)
+		var/obj/item/visor = user.glasses
+		if(visor) //make your hud fuck off
+			if(istype(visor,/obj/item/clothing/glasses/night/yautja))
```
Comment (harryob) [inline code/modules/cm_preds/yaut_mask.dm:162]:
```suggestion
			if(istype(visor, /obj/item/clothing/glasses/night/yautja))
```

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/cm_preds/yaut_mask.dm [hunk 6]
```diff
@@ -91,13 +113,13 @@
 	if(src != M.wear_mask) //sanity
 		to_chat(M, SPAN_WARNING("You must wear \the [src]!"))
 		return
-	var/obj/item/clothing/gloves/yautja/Y = M.gloves //Doesn't actually reduce power, but needs the bracers anyway.
+	var/obj/item/clothing/gloves/yautja/Y = M.gloves
```
Comment (BeagleGaming1) [inline code/modules/cm_preds/yaut_mask.dm:116]: single letter var

File: code/modules/cm_preds/yaut_mask.dm [hunk 7]
```diff
@@ -91,13 +113,13 @@
...
 	if(!Y || !istype(Y))
 		to_chat(M, SPAN_WARNING("You must be wearing your bracers, as they have the power source."))
 		return
 	var/obj/item/G = M.glasses
 	if(G)
-		if(!istype(G,/obj/item/clothing/glasses/night/yautja) && !istype(G,/obj/item/clothing/glasses/meson/yautja) && !istype(G,/obj/item/clothing/glasses/thermal/yautja))
+		if(!istype(G, /obj/item/clothing/glasses/night/yautja))
```
Comment (BeagleGaming1) [inline code/modules/cm_preds/yaut_mask.dm:122]: single letter var

**Review by BeagleGaming1 — CHANGES_REQUESTED** [review]

File: code/modules/cm_preds/yaut_mask.dm [hunk 8]
```diff
@@ -82,79 +104,77 @@
...
+	if(!HAS_TRAIT(user, TRAIT_YAUTJA_TECH) && !user.hunter_data.thralled)
+		to_chat(user, SPAN_WARNING("You have no idea how to work this thing!"))
 		return
-	if(src != M.wear_mask) //sanity
-		to_chat(M, SPAN_WARNING("You must wear \the [src]!"))
+	if(src != user.wear_mask) //sanity
+		to_chat(user, SPAN_WARNING("You must wear \the [src]!"))
 		return
-	var/obj/item/clothing/gloves/yautja/Y = M.gloves //Doesn't actually reduce power, but needs the bracers anyway.
+	var/obj/item/clothing/gloves/yautja/Y = user.gloves
```
Comment (BeagleGaming1) [inline code/modules/cm_preds/yaut_mask.dm]: single letter var

File: code/modules/cm_preds/yaut_mask.dm [hunk 9]
```diff
@@ -82,79 +104,77 @@
...
 	if(!Y || !istype(Y))
-		to_chat(M, SPAN_WARNING("You must be wearing your bracers, as they have the power source."))
+		to_chat(user, SPAN_WARNING("You must be wearing your bracers, as they have the power source."))
 		return
-	var/obj/item/G = M.glasses
+	var/obj/item/G = user.glasses
```
Comment (BeagleGaming1) [inline code/modules/cm_preds/yaut_mask.dm]: single letter var

**Review by BeagleGaming1 — APPROVED** [review]
Code good

File: code/modules/cm_preds/yaut_mask.dm [hunk 10]
```diff
@@ -65,8 +63,32 @@
...
+	STOP_PROCESSING(SSobj, src)
 	return ..()
 
+/obj/item/clothing/mask/gas/yautja/process()
+	if(!ishuman(loc))
+		return PROCESS_KILL
+	var/mob/living/carbon/human/human_holder = loc
+
+	if(current_goggles && !drain_power(human_holder, 3))
+		to_chat(human_holder, SPAN_WARNING("Your bracers lack sufficient power to operate the visior."))
```
Comment (BeagleGaming1) [inline code/modules/cm_preds/yaut_mask.dm:75]:
```suggestion
		to_chat(human_holder, SPAN_WARNING("Your bracers lack sufficient power to operate the visor."))
```

**Pull request merged**